src/frontend/src/instance/jaeger.rs

```diff
@@ -472,23 +483,64 @@ fn json_tag_filters(
     Ok(filters)
 }
 
-fn flatten_tag_filters(tags: HashMap<String, JsonValue>) -> ServerResult<Vec<Expr>> {
+fn flatten_tag_filters(
+    tags: HashMap<String, JsonValue>,
+    col_names: &HashSet<String>,
+) -> ServerResult<Vec<Expr>> {
+    /// Macro to check if span_key or resource_key exists in col_names and create an expression.
+    /// If neither exists, logs a warning and returns None.
+    macro_rules! check_col_and_build_expr {
+        ($span_key:expr, $resource_key:expr, $key:expr, $col_names:expr, $expr_builder:expr) => {{
+            if $col_names.contains(&$span_key) {
+                return Some($expr_builder($span_key));
+            }
+            if $col_names.contains(&$resource_key) {
+                return Some($expr_builder($resource_key));
+            }
+            warn!("tag key {} not found in table columns", $key);
+            None
+        }};
+    }
```

Comment:
Is it possible to implement this as a function? We need to avoid using macro where its possible.